### ISSUE TITLE
Task 393990 panicked

### ISSUE THREAD

**Issue by apozeus** [meilisearch/meilisearch#4002]
"Meilisearch ApiException: Http Status: 500 - Message: task 393990 panicked - Code: internal - Type: internal - Link: [https://docs.meilisearch.com/errors#internal"](https://docs.meilisearch.com/errors#internal%22)

**To Reproduce**
1. Search "Wedding ceremony 結婚式"
2. Error 500 occurs

**Meilisearch version:**
v1.3 Meilisearch cloud

**Comment by Kerollmops** [comment]
Hey @apozeus 👋

Thank you very much for your report. I will investigate this panic. I have a snapshot of a database with the same issue. I hope it will be enough to debug and fix that 🤞

**Comment by Kerollmops** [comment]
Hey @apozeus again 👋 

I am sorry, but do you accept that we use a copy of your Meilisearch data on the Cloud to debug? And if so, please communicate to me the URL of your instance here or via email. (you can find it on my GitHub profile).

**Comment by obrhoff** [comment]
Seeing a similar issue here. Is it related?

```
unaccepted status code found: 500 expected: [200], MeilisearchApiError Message: Inside `.queries[79]`: task 508094 panicked, Code: internal, Type: internal, Link: https://docs.meilisearch.com/errors#internal (path "POST /multi-search" with method "MultiSearch")
```

**Comment by Kerollmops** [comment]
Hey @obrhoff 👋

Thank you for reporting this error. Do you use the self-hosted version of Meilisearch? And if so, can you give me the log shown when this error is triggered please?

**Comment by obrhoff** [comment]
I just gave it a try and found this in the logs. ```thread 'tokio-runtime-worker' panicked at 'The ranking rule fid did not sort its bucket exhaustively', milli/src/search/new/bucket_sort.rs:162:13```

**Comment by Kerollmops** [comment]
Fixed by #4013.
